### PR TITLE
Extraction of storage_options from URL

### PR DESCRIPTION
**Pull request by manjuransari**
PR for issue #691

### CODE REVIEW

File: petastorm/fs_utils.py
Comment (selitvin) [inline petastorm/fs_utils.py:135]: Please don't use private methods of a third party library.

File: petastorm/fs_utils.py
Comment (selitvin) [inline petastorm/fs_utils.py:138]: Why do we construct `cls` twice? (lines 138 and 137)

File: petastorm/fs_utils.py
Comment (selitvin) [inline petastorm/fs_utils.py:217]: Please add a docstring and a unit test.

**Review by selitvin — CHANGES_REQUESTED** [review]
Is it possible to add some unittests to cover new functionality?

File: petastorm/fs_utils.py
Comment (manjuransari) [reply petastorm/fs_utils.py:135]: All third party libraries being used by fsspec implement class AbstractFileSystem. AbstractFileSystem class have this method which either developer can over ride according to his/her needs. More info: https://filesystem-spec.readthedocs.io/en/latest/developer.html

File: petastorm/fs_utils.py
Comment (manjuransari) [reply petastorm/fs_utils.py:138]: typo. Will update the code.

File: petastorm/fs_utils.py
Comment (manjuransari) [reply petastorm/fs_utils.py:32]: Its both clean up and change for handling the url that has storage options within them. All third party libraries being used by fsspec implement class AbstractFileSystem. AbstractFileSystem class have this method which either developer can over ride according to his/her needs. More info: https://filesystem-spec.readthedocs.io/en/latest/developer.html

File: petastorm/fs_utils.py
Comment (manjuransari) [reply petastorm/fs_utils.py:138]: typo. Updated changes.

File: petastorm/fs_utils.py
Comment (manjuransari) [reply petastorm/fs_utils.py]: Removed this method. Using the same method from fsspec library.

**Review by selitvin — COMMENTED** [review]
All looks good to me. One last ask: please update docs/release-notes.rst with the description of your change and then we'll land.

Thank you for your contribution!

**Review by selitvin — APPROVED** [review]

**Comment by manjuransari** [comment]
@selitvin can you please review the request.

**Comment by manjuransari** [comment]
@selitvin can you please review the changes.

**Review by selitvin — APPROVED** [review]
Sorry for the delay - was out for a vacation. Thanks for the contribution.

**Comment by manjuransari** [comment]
@selitvin can you review and approve the running workflows. It says first time contributor need maintainer to approve.

**Comment by selitvin** [comment]
Seems like flake8 fails. Not sure why is that. Will try looking at it today or tomorrow - unless you know what's going on?

**Comment by manjuransari** [comment]
Don't know why the flake8 fails.

**Comment by selitvin** [comment]
Trying to fix flake8 failure here: #701.

File: petastorm/fs_utils.py
Comment (selitvin) [inline petastorm/fs_utils.py:143]:
```suggestion
            self._filesystem_factory = lambda: cls(**options)  # pylint: disable=unnecessary-lambda
```

**Comment by manjuransari** [comment]
@selitvin build/test complete. Waiting for merge.

**Review by chongxiaoc — APPROVED** [review]

**Pull request merged**
